tests/tests_data/streaming/test_dataset.py

```diff
@@ -556,66 +593,101 @@ def test_resumable_dataset_single_worker(tmpdir):
     cache.done()
     cache.merge()
 
-    assert len([f for f in os.listdir(tmpdir) if f.endswith(".bin")]) == 50
+    assert len([f for f in os.listdir(data_dir) if f.endswith(".bin")]) == 50
 
-    dataset = StreamingDataset(input_dir=str(tmpdir), item_loader=TokensLoader(block_size), shuffle=True)
+    dataset = EmulateS3StreamingDataset(
+        input_dir=RemoteDir(cache_dir, data_dir), item_loader=TokensLoader(block_size), shuffle=True
+    )
```

Comment:
Maybe we should explicitly set a low `checkpoint_interval` here (to avoid flakiness?) I see below we are sleeping 0.1 seconds before we assert the state dict.

Reply:
This won't help much as it is already triggered by moving across chunks. The sleep are mostly to ensure the workers had enough time to run.